pilot/pkg/model/network_test.go

```diff
@@ -81,60 +83,60 @@ func TestGatewayHostnames(t *testing.T) {
 			t.Fatalf("unexpected network: %v", gateways)
 		}
 	})
-	t.Run("re-resolve after TTL", func(t *testing.T) {
-		if testing.Short() {
-			t.Skip()
-		}
-		// after the update, we should see the next gateway. Since TTL is low we don't know the exact IP, but we know it should change from
-		// the original
-		retry.UntilOrFail(t, func() bool {
-			return !reflect.DeepEqual(env.NetworkManager.AllGateways(), gateways)
+
+	if !testing.Short() {
```

Comment:
We have to wait seconds for these tests since DNS TTL is in seconds

Reply:
Using testing.Short is not appropriate in Istio IMO since we don't use it anywhere else so it will never be set. We should strive to make tests not timing dependent

Reply:
> Can we just make the ttl 0? then it will be set to `MinGatewayTTL`?

That would lose the converage of [`minimalTTL`](https://github.com/istio/istio/pull/44384/files#diff-0af799e3ed1e301215053a27e717344086e1410e0dbd46c05c25d9979175ce4cR504-R506). If you think timing is more important than coverage, I'd be happy to change it

Reply:
There is a PR trying to do that([#44429](https://github.com/istio/istio/pull/44429/files))

Reply:
@howardjohn Any thoughts? Would like to get this into 1.18